lib/gui.py

```diff
@@ -63,11 +75,39 @@ def on_submit():
             # Set the label text to the password generated
             password = generator.gen(int(user_input.get()))
             password_label.config(text=f"Password generated: {password}")
-            copy_button = tk.Button(password_generator_window, text='Copy to clipboard',
-                                    command=lambda: copy_to_clipboard(password))
-            copy_button.pack()
-        else:
-            password_label.config(text="Invalid input provided!")  # Update the label text
+            global copy_button
```

Comment:
Why is this global here exactly?

Reply:
Because I wanted that button to dissapear when user provide wrong input. Without that global you'll get
UnboundLocalError: local variable 'copy_button' referenced before assignment
else in this code don't see that button

Reply:
Why not declare the copy_button in the higher scope (outside the if block), put it in the grid if the check passed and don't do anything in the else block?

Reply:
I don't like the use of global. It really makes it not obvious where the code comes from